<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Figure 1 Vignette â€¢ loopBouquetPlotDocumentation</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Figure 1 Vignette">
<meta property="og:description" content="loopBouquetPlotDocumentation">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">loopBouquetPlotDocumentation</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BrainComparativeEpigenome.html">`loopBouquetPlot` for Brain Comparative Epigenome</a>
    </li>
    <li>
      <a href="../articles/Fig1.html">Figure 1 Vignette</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jianhong/loopBouquetPlotDocumentation" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Figure 1 Vignette</h1>
                        <h4 data-toc-skip class="author">Jianhong
Ou</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jianhong/loopBouquetPlotDocumentation/blob/HEAD/vignettes/Fig1.Rmd" class="external-link"><code>vignettes/Fig1.Rmd</code></a></small>
      <div class="hidden name"><code>Fig1.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>loopBouquetPlot</code> is a new method introduced by
<strong>trackViewer</strong> package to visualize genomic interactions
along with annotation for NGS dataset such as HiC, HiChIP, PLAC-seq,
ChIA-PET, and HiCAR data.</p>
<p>The Fig 1 is the showcase for the <code>loopBouquetPlot</code>
compared to the heatmaps and links plot. Additional showcase are
available at <a href="BrainComparativeEpigenome.html"><code>loopBouquetPlot</code> for
Brain Comparative Epigenome</a>. There are several steps to repeat the
<code>loopBouquetPlot</code>:</p>
<ol style="list-style-type: decimal">
<li><p>Prepare the annotations</p></li>
<li><p>Import the called interactions</p></li>
<li><p>Import the ATAC-seq signals if available</p></li>
<li><p>Plot the data by <code>loopBouquetPlot</code> function.</p></li>
</ol>
</div>
<div class="section level2">
<h2 id="load-libraries">Load libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">trackViewer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">InteractionSet</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Dr.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicFeatures" class="external-link">GenomicFeatures</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomeInfoDb" class="external-link">GenomeInfoDb</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rmariadb.r-dbi.org" class="external-link">RMariaDB</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ChIPpeakAnno</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-annotation">Prepare annotation<a class="anchor" aria-label="anchor" href="#prepare-annotation"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## prepare the TxDb object,</span></span>
<span><span class="co">## or load the refseq gene annotation by library(TxDb.Drerio.UCSC.danRer11.refGene)</span></span>
<span><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="fu">GenomicFeatures</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromEnsembl.html" class="external-link">makeTxDbFromEnsembl</a></span><span class="op">(</span>release<span class="op">=</span><span class="fl">109</span>,</span>
<span>                                             organism<span class="op">=</span><span class="st">'Danio rerio'</span><span class="op">)</span></span>
<span><span class="co">## set the silencer coordinates</span></span>
<span><span class="va">silencer</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr5"</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fl">21795000</span>, <span class="fl">21800000</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## set data range</span></span>
<span><span class="va">range</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr5"</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fl">21100000</span>, <span class="fl">22100000</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## set the plot range</span></span>
<span><span class="va">fig1PlotRange</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">'chr5'</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fl">21300000</span>, <span class="fl">2.2e+07</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## prepare gene annotation</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">GenomicFeatures</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span></span>
<span><span class="fu">GenomeInfoDb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">genes</span>, <span class="va">range</span>, ignore.strand<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">id</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">$</span><span class="va">gene_id</span></span>
<span><span class="va">eid</span> <span class="op">&lt;-</span> <span class="fu">ChIPpeakAnno</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/xget.html" class="external-link">xget</a></span><span class="op">(</span><span class="va">id</span>, <span class="va">org.Dr.egENSEMBL2EG</span>, output <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span></span>
<span><span class="va">symbol</span> <span class="op">&lt;-</span> <span class="va">id</span></span>
<span><span class="va">symbol</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">eid</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">ChIPpeakAnno</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/xget.html" class="external-link">xget</a></span><span class="op">(</span><span class="va">eid</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">eid</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                                          <span class="va">org.Dr.egSYMBOL</span>,</span>
<span>                                          output <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">id</span></span>
<span></span>
<span><span class="va">feature.gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">genes</span>, <span class="va">range</span>, ignore.strand<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">label</span> <span class="op">&lt;-</span> <span class="va">symbol</span><span class="op">[</span><span class="va">feature.gr</span><span class="op">$</span><span class="va">gene_id</span><span class="op">]</span></span>
<span><span class="co"># remove genes without symbols to clean the plot</span></span>
<span><span class="va">feature.gr</span> <span class="op">&lt;-</span> <span class="va">feature.gr</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">'ENSDARG'</span>, <span class="va">feature.gr</span><span class="op">$</span><span class="va">label</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">## show selected gene symbols only</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">label</span><span class="op">[</span><span class="op">!</span><span class="va">feature.gr</span><span class="op">$</span><span class="va">label</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'tenm1'</span>, <span class="st">'smarca1'</span>, <span class="st">'sh2d1aa'</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">''</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="st">'gene'</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">col</span> <span class="op">&lt;-</span> <span class="fl">1</span> <span class="co"># set black color to uninterested genes</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">col</span><span class="op">[</span><span class="va">feature.gr</span><span class="op">$</span><span class="va">label</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'tenm1'</span>, <span class="st">'smarca1'</span>, <span class="st">'sh2d1aa'</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">5</span><span class="op">)</span> <span class="co">## set the colors for the given genes</span></span>
<span><span class="va">silencer</span><span class="op">$</span><span class="va">label</span> <span class="op">&lt;-</span> <span class="st">'silencer'</span></span>
<span><span class="va">silencer</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="st">'cRE'</span> <span class="co">## cis regulatory element</span></span>
<span><span class="va">silencer</span><span class="op">$</span><span class="va">col</span> <span class="op">&lt;-</span> <span class="fl">3</span> <span class="co">## set the color to green</span></span>
<span><span class="va">feature.gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">feature.gr</span>, <span class="va">silencer</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="subfigure-b-and-c">Subfigure <strong>b</strong> and <strong>c</strong><a class="anchor" aria-label="anchor" href="#subfigure-b-and-c"></a>
</h2>
<div class="section level3">
<h3 id="import-data-for-loopbouquetplot">Import data for <code>loopBouquetPlot</code><a class="anchor" aria-label="anchor" href="#import-data-for-loopbouquetplot"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## set the data folder, all data are available in the extdata folder of this package</span></span>
<span><span class="va">extdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, package<span class="op">=</span><span class="st">'loopBouquetPlotDocumentation'</span><span class="op">)</span></span>
<span><span class="co">## list all interaction BEDPE files</span></span>
<span><span class="op">(</span><span class="va">interactions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="va">extdata</span>,</span>
<span>                     <span class="st">"10k.2.sig3Dinteractions.bedpe"</span>,</span>
<span>                     full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/tmp/RtmpuJW8D5/temp_libpath3d6385c5b0c/loopBouquetPlotDocumentation/extdata/0dpa.10k.2.sig3Dinteractions.bedpe"</span></span>
<span><span class="co">## [2] "/tmp/RtmpuJW8D5/temp_libpath3d6385c5b0c/loopBouquetPlotDocumentation/extdata/4dpa.10k.2.sig3Dinteractions.bedpe"</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## import the loop for `loopBouquetPlot`</span></span>
<span><span class="va">loops_loopBouquet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">interactions</span>, <span class="kw">function</span><span class="op">(</span><span class="va">.ele</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co">## read bedpe as text file</span></span>
<span>  <span class="va">.loop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">.ele</span>, header <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="co">## create GInteractions object</span></span>
<span>  <span class="va">.loop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">.loop</span>, <span class="fu">InteractionSet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/GInteractions-class.html" class="external-link">GInteractions</a></span><span class="op">(</span></span>
<span>    <span class="co">## narrow the interaction region for better visualization effect</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">V1</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">V2</span><span class="op">+</span><span class="fl">2501</span>, <span class="va">V3</span><span class="op">-</span><span class="fl">2500</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">V4</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">V5</span><span class="op">+</span><span class="fl">2501</span>, <span class="va">V6</span><span class="op">-</span><span class="fl">2500</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="co">## set the score to log2 value to decrease the difference of signals</span></span>
<span>    score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">V8</span><span class="op">)</span></span>
<span>  <span class="op">)</span><span class="op">)</span></span>
<span>  <span class="co">## subset the data with given range</span></span>
<span>  <span class="va">.loop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">.loop</span>, <span class="va">range</span><span class="op">)</span></span>
<span>  <span class="co">## remove the interchromosal interactions</span></span>
<span>  <span class="va">.loop</span> <span class="op">&lt;-</span> <span class="va">.loop</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Pairs-class.html" class="external-link">first</a></span><span class="op">(</span><span class="va">.loop</span><span class="op">)</span><span class="op">)</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">range</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&amp;</span></span>
<span>                   <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Pairs-class.html" class="external-link">second</a></span><span class="op">(</span><span class="va">.loop</span><span class="op">)</span><span class="op">)</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">range</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="co">## create names for the data for each sample</span></span>
<span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">loops_loopBouquet</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">".10k.*$"</span>, <span class="st">" interactions"</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">interactions</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "0dpa interactions" "4dpa interactions"</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## list all teh ATAC-seq bigwig files</span></span>
<span><span class="op">(</span><span class="va">bws</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="va">extdata</span>,</span>
<span>            <span class="st">"bigWig"</span>,</span>
<span>            full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/tmp/RtmpuJW8D5/temp_libpath3d6385c5b0c/loopBouquetPlotDocumentation/extdata/0dpa.bigWig"</span></span>
<span><span class="co">## [2] "/tmp/RtmpuJW8D5/temp_libpath3d6385c5b0c/loopBouquetPlotDocumentation/extdata/4dpa.bigWig"</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## import the ATAC-seq signales</span></span>
<span><span class="co">## the ATAC-seq signals are HiCAR R2 read coverage</span></span>
<span><span class="va">bwsigs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">bws</span>, <span class="fu">trackViewer</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/trackViewer/man/importScore.html" class="external-link">importScore</a></span>, format<span class="op">=</span><span class="st">"BigWig"</span>, ranges<span class="op">=</span><span class="va">range</span><span class="op">)</span></span>
<span><span class="co">## create names for the data for each sample</span></span>
<span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bwsigs</span><span class="op">)</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">'.bigWig'</span>, <span class="st">' ATAC-seq'</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">bws</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "0dpa ATAC-seq" "4dpa ATAC-seq"</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## check the order of the samples are same for ATAC-seq signals and loops</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">' .*$'</span>, <span class="st">''</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bwsigs</span><span class="op">)</span><span class="op">)</span><span class="op">==</span></span>
<span>                <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">' .*$'</span>, <span class="st">''</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">loops_loopBouquet</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-the-data-by-loopbouquetplot-for-subfigure-b-and-c">Plot the data by <code>loopBouquetPlot</code> for subfigure
<strong>b</strong> and <strong>c</strong><a class="anchor" aria-label="anchor" href="#plot-the-data-by-loopbouquetplot-for-subfigure-b-and-c"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> <span class="co">## set seed to control the plot layout</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">loops_loopBouquet</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co">## pdf(paste0(names(loops_loopBouquet)[i], '.pdf')) # if using terminal</span></span>
<span>  <span class="fu">trackViewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/loopBouquetPlot.html" class="external-link">loopBouquetPlot</a></span><span class="op">(</span><span class="va">loops_loopBouquet</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                               <span class="va">fig1PlotRange</span>,</span>
<span>                               <span class="va">feature.gr</span>,</span>
<span>                               atacSig<span class="op">=</span><span class="va">bwsigs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                               lwd.backbone<span class="op">=</span><span class="fl">2</span>,</span>
<span>                               col.backbone<span class="op">=</span><span class="st">'gray'</span>,</span>
<span>                               col.backbone_background <span class="op">=</span> <span class="st">'gray70'</span>,</span>
<span>                               lwd.gene <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                               show_coor <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                               coor_tick_unit <span class="op">=</span> <span class="fl">1e4</span>,</span>
<span>                               coor_mark_interval<span class="op">=</span><span class="fl">2e5</span>,</span>
<span>                               length.arrow<span class="op">=</span><span class="fl">0.1</span>,</span>
<span>                               show_edges <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span>label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">loops_loopBouquet</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>,</span>
<span>            x<span class="op">=</span><span class="fl">.15</span>, y<span class="op">=</span><span class="fl">.9</span><span class="op">)</span></span>
<span>  <span class="co">## dev.off() # if using terminal</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="Fig1_files/figure-html/loopBouquetPlot-1.png" width="700"><img src="Fig1_files/figure-html/loopBouquetPlot-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="subfigure-a">Subfigure <strong>a</strong><a class="anchor" aria-label="anchor" href="#subfigure-a"></a>
</h2>
<p>There are several steps to repeat the heatmaps and links plot:</p>
<ol style="list-style-type: decimal">
<li><p>Prepare the annotations</p></li>
<li><p>Import the ATAC-seq signals</p></li>
<li><p>Import the called interactions for link tracks</p></li>
<li><p>Import the interactions for heatmap tracks</p></li>
<li><p>Plot the data by <code>viewTracks</code> function.</p></li>
</ol>
<div class="section level3">
<h3 id="import-data-for-subfigure-a">Import data for subfigure <strong>a</strong><a class="anchor" aria-label="anchor" href="#import-data-for-subfigure-a"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## import the interactions as 'track' objects</span></span>
<span><span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">interactions</span>, <span class="kw">function</span><span class="op">(</span><span class="va">.ele</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">.loop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">.ele</span>, header <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="va">.loop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">.loop</span>, <span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/GInteractions-class.html" class="external-link">GInteractions</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">V1</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">V2</span><span class="op">+</span><span class="fl">2501</span>, <span class="va">V3</span><span class="op">-</span><span class="fl">2500</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">V4</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">V5</span><span class="op">+</span><span class="fl">2501</span>, <span class="va">V6</span><span class="op">-</span><span class="fl">2500</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">V8</span><span class="op">)</span></span>
<span>  <span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">.loop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/gi2track.html" class="external-link">gi2track</a></span><span class="op">(</span><span class="va">.loop</span><span class="op">)</span></span>
<span>  <span class="co">## set the tracktype to 'link' to plot as links</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">setTrackStyleParam</a></span><span class="op">(</span><span class="va">.loop</span>, <span class="st">"tracktype"</span>, <span class="st">"link"</span><span class="op">)</span></span>
<span>  <span class="va">.loop</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">loops</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">".10k.*$"</span>, <span class="st">" interactions"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">interactions</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## get the hic files. Here we use the prepared GInteraction objects</span></span>
<span><span class="op">(</span><span class="va">hics</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="va">extdata</span>,</span>
<span>             <span class="st">"10000.ginteractions.rds"</span>,</span>
<span>             full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/tmp/RtmpuJW8D5/temp_libpath3d6385c5b0c/loopBouquetPlotDocumentation/extdata/MAPS_signals_0dpa.10000.ginteractions.rds"</span></span>
<span><span class="co">## [2] "/tmp/RtmpuJW8D5/temp_libpath3d6385c5b0c/loopBouquetPlotDocumentation/extdata/MAPS_signals_4dpa.10000.ginteractions.rds"</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## set the sample names</span></span>
<span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hics</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"MAPS_signals_(.*?).10000.ginteractions.rds"</span>, <span class="st">"\\1 heatmap"</span>,</span>
<span>                    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">hics</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "0dpa heatmap" "4dpa heatmap"</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">hics</span>, <span class="kw">function</span><span class="op">(</span><span class="va">.ele</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co">## The RDS files were generated by trackViewer function `importGInteractions`</span></span>
<span>  <span class="co">## from the HIC file. Because the MAPS signal are values normalized by</span></span>
<span>  <span class="co">## fitting models, we set the matrixType to 'observed' and normalization to 'NONE'</span></span>
<span>  <span class="co">## range.ext are ranges with extention to include all the intrachromosal interactions</span></span>
<span>  <span class="co"># range.ext &lt;- range</span></span>
<span>  <span class="co"># start(range.ext) &lt;- start(range.ext) - 10000000</span></span>
<span>  <span class="co"># end(range.ext) &lt;- end(range.ext) + 10000000</span></span>
<span>  <span class="co"># importGInteractions(.ele, ranges = range.ext,</span></span>
<span>  <span class="co">#                     format = "hic",</span></span>
<span>  <span class="co">#                     resolution = 10000,</span></span>
<span>  <span class="co">#                     out = "GInteractions",</span></span>
<span>  <span class="co">#                     normalization = "NONE",</span></span>
<span>  <span class="co">#                     matrixType = "observed")</span></span>
<span>  <span class="va">.ele</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">.ele</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="co">## check the order of the samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">" .*$"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gis</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">" .*$"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">loops</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## create heatmap signals</span></span>
<span><span class="va">heatmaps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mapply.html" class="external-link">mapply</a></span><span class="op">(</span><span class="va">gis</span>, <span class="va">interactions</span>, FUN<span class="op">=</span><span class="kw">function</span><span class="op">(</span><span class="va">.gi</span>, <span class="va">.ele</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">.loop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">.ele</span>, header <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="va">.loop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">.loop</span>, <span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/GInteractions-class.html" class="external-link">GInteractions</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">V1</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">V2</span><span class="op">+</span><span class="fl">2501</span>, <span class="va">V3</span><span class="op">-</span><span class="fl">2500</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">V4</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">V5</span><span class="op">+</span><span class="fl">2501</span>, <span class="va">V6</span><span class="op">-</span><span class="fl">2500</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">V8</span><span class="op">)</span></span>
<span>  <span class="op">)</span><span class="op">)</span></span>
<span>  <span class="co">## highlight the called interactions by add color the border</span></span>
<span>  <span class="va">.gi</span><span class="op">$</span><span class="va">border_color</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span>  <span class="va">ol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">.gi</span>, <span class="va">.loop</span><span class="op">)</span></span>
<span>  <span class="va">.gi</span><span class="op">$</span><span class="va">border_color</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">ol</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"black"</span></span>
<span>  <span class="co">## create 'track' object from GInteraction object</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/gi2track.html" class="external-link">gi2track</a></span><span class="op">(</span><span class="va">.gi</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="co">## create gene track object</span></span>
<span><span class="va">trs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/geneTrack.html" class="external-link">geneTrack</a></span><span class="op">(</span><span class="va">id</span>, <span class="va">txdb</span>, <span class="va">symbol</span>, asList <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">trs</span><span class="op">$</span><span class="va">dat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span> <span class="co">## make sure the seqnames are in same style</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-the-subfigure-a-by-viewtracks">Plot the subfigure <strong>a</strong> by viewTracks<a class="anchor" aria-label="anchor" href="#plot-the-subfigure-a-by-viewtracks"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create the plot style by default algorithm</span></span>
<span><span class="va">op</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/optimizeStyle.html" class="external-link">optimizeStyle</a></span><span class="op">(</span></span>
<span>  trackList <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackList-class.html" class="external-link">trackList</a></span><span class="op">(</span><span class="va">trs</span>, </span>
<span>                        <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">bwsigs</span><span class="op">)</span>,   <span class="co">## reverse the signal</span></span>
<span>                        <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">loops</span><span class="op">)</span>,    <span class="co">## to keep 0 dpa samples</span></span>
<span>                        <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">heatmaps</span><span class="op">)</span>, <span class="co">## in top</span></span>
<span>                        <span class="co">## set the height for each track</span></span>
<span>                        heightDist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">bwsigs</span><span class="op">)</span>,</span>
<span>                                       <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">loops</span><span class="op">)</span>,</span>
<span>                                       <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">heatmaps</span><span class="op">)</span><span class="op">*</span><span class="fl">8</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tL</span> <span class="op">&lt;-</span> <span class="va">op</span><span class="op">$</span><span class="va">tracks</span></span>
<span><span class="va">sty</span> <span class="op">&lt;-</span> <span class="va">op</span><span class="op">$</span><span class="va">style</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">'ATAC-seq'</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tL</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span><span class="co">#R2 ATAC-seq signal</span></span>
<span>  <span class="co">## set the ylim. The raw range is [0,1]</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">setTrackStyleParam</a></span><span class="op">(</span><span class="va">tL</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="st">"ylim"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.15</span>, <span class="fl">.8</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="co">## set the y-axis to the right side</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">setTrackYaxisParam</a></span><span class="op">(</span><span class="va">tL</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="st">"main"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="co">## plot the y-axis lables</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">setTrackYaxisParam</a></span><span class="op">(</span><span class="va">tL</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="st">"label"</span>, <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="co">## draw the y-axis</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">setTrackYaxisParam</a></span><span class="op">(</span><span class="va">tL</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="st">"draw"</span>, <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">## set color keys for the link tracks</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">'interactions'</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tL</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">setTrackStyleParam</a></span><span class="op">(</span><span class="va">tL</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="st">"breaks"</span>,</span>
<span>                     <span class="co">## set the breaks from min to max</span></span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>from<span class="op">=</span><span class="fl">2</span>, to<span class="op">=</span><span class="fl">6.5</span>, by<span class="op">=</span><span class="fl">.5</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">setTrackStyleParam</a></span><span class="op">(</span><span class="va">tL</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="st">"color"</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lightblue"</span>, <span class="st">"yellow"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">setTrackYaxisParam</a></span><span class="op">(</span><span class="va">tL</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="st">"draw"</span>, <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">## set color keys for the heatmaps</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">'heatmap'</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tL</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">setTrackStyleParam</a></span><span class="op">(</span><span class="va">tL</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="st">"breaks"</span>, </span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>from<span class="op">=</span><span class="fl">0</span>, to<span class="op">=</span><span class="fl">5</span>, by<span class="op">=</span><span class="fl">.5</span><span class="op">)</span>, <span class="fl">8.5</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">setTrackStyleParam</a></span><span class="op">(</span><span class="va">tL</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="st">"color"</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#FDF8F5"</span>, <span class="st">"#EA7556"</span>, <span class="st">"#4D0E0B"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">##</span></span>
<span><span class="co">## pdf('Fig1a.pdf') # if using terminal</span></span>
<span><span class="fu">trackViewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/viewTracks.html" class="external-link">viewTracks</a></span><span class="op">(</span><span class="va">tL</span>, viewerStyle <span class="op">=</span> <span class="va">sty</span>,</span>
<span>                        gr<span class="op">=</span><span class="va">fig1PlotRange</span>, autoOptimizeStyle <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## set the TADs to plot with the heatmap</span></span>
<span><span class="va">tad</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr5"</span>,</span>
<span>               <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">21060000</span>, <span class="fl">21220000</span>, <span class="fl">21326000</span>, <span class="fl">21890000</span>, <span class="fl">22000000</span><span class="op">)</span>,</span>
<span>                       <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">21130000</span>, <span class="fl">21290000</span>, <span class="fl">21860000</span>, <span class="fl">21960000</span>, <span class="fl">22090000</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## add the annotations to the heatmap</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">'heatmap'</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tL</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co">## add TADs as dashed line (lty=2)</span></span>
<span>  <span class="fu">trackViewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/addInteractionAnnotation.html" class="external-link">addInteractionAnnotation</a></span><span class="op">(</span><span class="va">tad</span>, <span class="va">i</span>, <span class="va">grid.lines</span>,</span>
<span>                                        gp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"black"</span>,</span>
<span>                                                lwd<span class="op">=</span><span class="fl">1</span>, lty<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="co">## add the guide line for the silencer with slope -1 with lwd=3</span></span>
<span>  <span class="fu">trackViewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/addInteractionAnnotation.html" class="external-link">addInteractionAnnotation</a></span><span class="op">(</span><span class="op">-</span><span class="fl">21795000</span>, <span class="va">i</span>,</span>
<span>                                        gp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>col <span class="op">=</span> <span class="st">"#009E73"</span>,</span>
<span>                                                lwd<span class="op">=</span><span class="fl">3</span>, lty<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="Fig1_files/figure-html/viewTracks-1.png" width="672"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## dev.off() # if using terminal</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R Under development (unstable) (2024-02-07 r85873)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] grid      stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ChIPpeakAnno_3.37.4         RMariaDB_1.3.1             </span></span>
<span><span class="co">##  [3] GenomicFeatures_1.55.3      org.Dr.eg.db_3.18.0        </span></span>
<span><span class="co">##  [5] AnnotationDbi_1.65.2        InteractionSet_1.31.0      </span></span>
<span><span class="co">##  [7] SummarizedExperiment_1.33.3 Biobase_2.63.0             </span></span>
<span><span class="co">##  [9] MatrixGenerics_1.15.0       matrixStats_1.2.0          </span></span>
<span><span class="co">## [11] trackViewer_1.39.11         GenomicRanges_1.55.2       </span></span>
<span><span class="co">## [13] GenomeInfoDb_1.39.6         IRanges_2.37.1             </span></span>
<span><span class="co">## [15] S4Vectors_0.41.3            BiocGenerics_0.49.1        </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.4.0            BiocIO_1.13.0            bitops_1.0-7            </span></span>
<span><span class="co">##   [4] filelock_1.0.3           tibble_3.2.1             graph_1.81.0            </span></span>
<span><span class="co">##   [7] XML_3.99-0.16.1          rpart_4.1.23             lifecycle_1.0.4         </span></span>
<span><span class="co">##  [10] httr2_1.0.0              lattice_0.22-5           ensembldb_2.27.1        </span></span>
<span><span class="co">##  [13] MASS_7.3-60.2            backports_1.4.1          magrittr_2.0.3          </span></span>
<span><span class="co">##  [16] Hmisc_5.1-1              sass_0.4.8               rmarkdown_2.25          </span></span>
<span><span class="co">##  [19] jquerylib_0.1.4          yaml_2.3.8               plotrix_3.8-4           </span></span>
<span><span class="co">##  [22] Gviz_1.47.1              DBI_1.2.1                RColorBrewer_1.1-3      </span></span>
<span><span class="co">##  [25] lubridate_1.9.3          abind_1.4-5              zlibbioc_1.49.0         </span></span>
<span><span class="co">##  [28] purrr_1.0.2              AnnotationFilter_1.27.0  biovizBase_1.51.0       </span></span>
<span><span class="co">##  [31] RCurl_1.98-1.14          nnet_7.3-19              VariantAnnotation_1.49.6</span></span>
<span><span class="co">##  [34] rappdirs_0.3.3           GenomeInfoDbData_1.2.11  grImport_0.9-7          </span></span>
<span><span class="co">##  [37] pkgdown_2.0.7            codetools_0.2-19         DelayedArray_0.29.1     </span></span>
<span><span class="co">##  [40] xml2_1.3.6               tidyselect_1.2.0         futile.logger_1.4.3     </span></span>
<span><span class="co">##  [43] universalmotif_1.21.2    BiocFileCache_2.11.1     base64enc_0.1-3         </span></span>
<span><span class="co">##  [46] GenomicAlignments_1.39.2 jsonlite_1.8.8           multtest_2.59.0         </span></span>
<span><span class="co">##  [49] Formula_1.2-5            survival_3.5-7           systemfonts_1.0.5       </span></span>
<span><span class="co">##  [52] tools_4.4.0              progress_1.2.3           ragg_1.2.7              </span></span>
<span><span class="co">##  [55] strawr_0.0.91            Rcpp_1.0.12              glue_1.7.0              </span></span>
<span><span class="co">##  [58] gridExtra_2.3            SparseArray_1.3.3        xfun_0.42               </span></span>
<span><span class="co">##  [61] dplyr_1.1.4              formatR_1.14             fastmap_1.1.1           </span></span>
<span><span class="co">##  [64] latticeExtra_0.6-30      rhdf5filters_1.15.2      fansi_1.0.6             </span></span>
<span><span class="co">##  [67] digest_0.6.34            timechange_0.3.0         R6_2.5.1                </span></span>
<span><span class="co">##  [70] textshaping_0.3.7        colorspace_2.1-0         jpeg_0.1-10             </span></span>
<span><span class="co">##  [73] dichromat_2.0-0.1        biomaRt_2.59.1           RSQLite_2.3.5           </span></span>
<span><span class="co">##  [76] tidyr_1.3.1              utf8_1.2.4               generics_0.1.3          </span></span>
<span><span class="co">##  [79] data.table_1.15.0        rtracklayer_1.63.0       prettyunits_1.2.0       </span></span>
<span><span class="co">##  [82] httr_1.4.7               htmlwidgets_1.6.4        S4Arrays_1.3.3          </span></span>
<span><span class="co">##  [85] regioneR_1.35.0          pkgconfig_2.0.3          gtable_0.3.4            </span></span>
<span><span class="co">##  [88] blob_1.2.4               XVector_0.43.1           htmltools_0.5.7         </span></span>
<span><span class="co">##  [91] RBGL_1.79.0              ProtGenerics_1.35.2      scales_1.3.0            </span></span>
<span><span class="co">##  [94] png_0.1-8                knitr_1.45               lambda.r_1.2.4          </span></span>
<span><span class="co">##  [97] rstudioapi_0.15.0        rjson_0.2.21             checkmate_2.3.1         </span></span>
<span><span class="co">## [100] curl_5.2.0               cachem_1.0.8             rhdf5_2.47.3            </span></span>
<span><span class="co">## [103] stringr_1.5.1            parallel_4.4.0           foreign_0.8-86          </span></span>
<span><span class="co">## [106] restfulr_0.0.15          desc_1.4.3               pillar_1.9.0            </span></span>
<span><span class="co">## [109] vctrs_0.6.5              dbplyr_2.4.0             cluster_2.1.6           </span></span>
<span><span class="co">## [112] htmlTable_2.4.2          evaluate_0.23            VennDiagram_1.7.3       </span></span>
<span><span class="co">## [115] cli_3.6.2                compiler_4.4.0           futile.options_1.0.1    </span></span>
<span><span class="co">## [118] Rsamtools_2.19.3         rlang_1.1.3              crayon_1.5.2            </span></span>
<span><span class="co">## [121] interp_1.1-6             fs_1.6.3                 stringi_1.8.3           </span></span>
<span><span class="co">## [124] deldir_2.0-2             BiocParallel_1.37.0      munsell_0.5.0           </span></span>
<span><span class="co">## [127] Biostrings_2.71.2        lazyeval_0.2.2           Matrix_1.6-5            </span></span>
<span><span class="co">## [130] BSgenome_1.71.2          hms_1.1.3                bit64_4.0.5             </span></span>
<span><span class="co">## [133] ggplot2_3.4.4            Rhdf5lib_1.25.1          KEGGREST_1.43.0         </span></span>
<span><span class="co">## [136] highr_0.10               igraph_2.0.1.1           memoise_2.0.1           </span></span>
<span><span class="co">## [139] bslib_0.6.1              bit_4.0.5</span></span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jianhong Ou.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
