<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>`loopBouquetPlot` example for Human HiC and ChIA-PET data â€¢ loopBouquetPlotDocumentation</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="`loopBouquetPlot` example for Human HiC and ChIA-PET data">
<meta property="og:description" content="loopBouquetPlotDocumentation">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">loopBouquetPlotDocumentation</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Fig1.html">Figure 1 Vignette</a>
    </li>
    <li>
      <a href="../articles/HumanSample.html">`loopBouquetPlot` example for Human HiC and ChIA-PET data</a>
    </li>
    <li>
      <a href="../articles/MAPS.html">`loopBouquetPlot` example for loops caller comparison</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jianhong/loopBouquetPlotDocumentation" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>
<code>loopBouquetPlot</code> example for Human
HiC and ChIA-PET data</h1>
                        <h4 data-toc-skip class="author">Jianhong
Ou</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jianhong/loopBouquetPlotDocumentation/blob/HEAD/vignettes/HumanSample.Rmd" class="external-link"><code>vignettes/HumanSample.Rmd</code></a></small>
      <div class="hidden name"><code>HumanSample.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>To illustrate how <code>loopBouquetPlot</code> can be used to help
the comparison of different types of datasets, the humnan cellline
data<span class="citation"><sup>2</sup></span> will be plotted here.</p>
<p>There are several steps to repeat the plots:</p>
<ol style="list-style-type: decimal">
<li><p>Prepare the annotations</p></li>
<li><p>Import interactions</p></li>
<li><p>Import the ATAC-seq signals if available</p></li>
<li><p>Plot the data by <code>loopBouquetPlot</code> function.</p></li>
</ol>
</div>
<div class="section level2">
<h2 id="load-libraries">Load libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">trackViewer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">InteractionSet</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ChIPpeakAnno</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="case-1--plot-called-loops-around-adamts5-gene-from-hic-seq-data">Case 1. plot called loops around <em>ADAMTS5</em> gene from HiC-seq
data<a class="anchor" aria-label="anchor" href="#case-1--plot-called-loops-around-adamts5-gene-from-hic-seq-data"></a>
</h2>
<p>Chromatin loops for human cell lines were identified at kilobase
resolution using in situ Hi-C sequencing<span class="citation"><sup>1</sup></span>. In this use case, we will showcase
the capabilities of <code>loopBouquetPlot</code> for bin-based
loops.</p>
<div class="section level3">
<h3 id="import-interactions">Import interactions<a class="anchor" aria-label="anchor" href="#import-interactions"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## define the data file path</span></span>
<span><span class="va">extdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, </span>
<span>                       <span class="st">'GSE63525'</span>,</span>
<span>                       package<span class="op">=</span><span class="st">'loopBouquetPlotDocumentation'</span><span class="op">)</span></span>
<span><span class="va">readData</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">HiCCUPS_looplist_file</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">extdata</span>,</span>
<span>              <span class="va">HiCCUPS_looplist_file</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="co">## Use observed value/ expected value (donut)</span></span>
<span>  <span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">loops</span>, <span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/GInteractions-class.html" class="external-link">GInteractions</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">chr1</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">x1</span><span class="op">+</span><span class="fl">1</span>, <span class="va">x2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">chr2</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">y1</span><span class="op">+</span><span class="fl">1</span>, <span class="va">y2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">o</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">e_donut</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="co">## change the chromosome names style to UCSC (with prefix 'chr')</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">loops</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'UCSC'</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">loops</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">GM12878_gi</span> <span class="op">&lt;-</span> <span class="fu">readData</span><span class="op">(</span><span class="st">'GSE63525_GM12878_HiCCUPS_looplist.txt.gz'</span><span class="op">)</span></span>
<span><span class="va">IMR90_gi</span> <span class="op">&lt;-</span> <span class="fu">readData</span><span class="op">(</span><span class="st">'GSE63525_IMR90_HiCCUPS_looplist.txt.gz'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prepare-the-annotations">Prepare the annotations<a class="anchor" aria-label="anchor" href="#prepare-the-annotations"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs_feature</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span></span>
<span><span class="va">hs_feature</span><span class="op">$</span><span class="va">symbol</span> <span class="op">&lt;-</span> <span class="fu">ChIPpeakAnno</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/xget.html" class="external-link">xget</a></span><span class="op">(</span><span class="va">hs_feature</span><span class="op">$</span><span class="va">gene_id</span>, <span class="va">org.Hs.egSYMBOL</span><span class="op">)</span></span>
<span><span class="co">## define the plot regions</span></span>
<span><span class="va">selectMarkers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ADAMTS1'</span>, <span class="st">'ADAMTS5'</span>, <span class="st">'N6AMT1'</span><span class="op">)</span></span>
<span><span class="va">hs_range</span> <span class="op">&lt;-</span> <span class="va">hs_feature</span><span class="op">[</span><span class="va">hs_feature</span><span class="op">$</span><span class="va">symbol</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">selectMarkers</span><span class="op">]</span></span>
<span><span class="va">hs_rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">hs_range</span><span class="op">)</span></span>
<span><span class="co"># subset the data</span></span>
<span><span class="va">IMR90_gi_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">IMR90_gi</span>, <span class="va">hs_rd</span><span class="op">)</span></span>
<span><span class="va">GM12878_gi_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">GM12878_gi</span>, <span class="va">hs_rd</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## prepare the annotations</span></span>
<span><span class="va">feature.gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">hs_feature</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/interaction-accessors.html" class="external-link">regions</a></span><span class="op">(</span><span class="va">IMR90_gi_sub</span><span class="op">)</span>,</span>
<span>                                       <span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/interaction-accessors.html" class="external-link">regions</a></span><span class="op">(</span><span class="va">GM12878_gi_sub</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">feature.gr</span> <span class="op">&lt;-</span> <span class="va">feature.gr</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">hs_rd</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">label</span> <span class="op">&lt;-</span> <span class="va">feature.gr</span><span class="op">$</span><span class="va">symbol</span></span>
<span><span class="co">## not label the gene outside the range</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">label</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span><span class="op">&lt;</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">hs_rd</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span><span class="op">&gt;</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">hs_rd</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span><span class="co">## assign colors for the genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">7</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span>, replace<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## set the annotation type to 'gene'</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="st">'gene'</span></span>
<span><span class="va">feature.gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 8 ranges and 5 metadata columns:</span></span>
<span><span class="co">##             seqnames            ranges strand |     gene_id      symbol</span></span>
<span><span class="co">##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">##   100616243    chr21 28326280-28326362      + |   100616243     MIR4759</span></span>
<span><span class="co">##       11096    chr21 28290231-28339439      - |       11096     ADAMTS5</span></span>
<span><span class="co">##      118421    chr21 29911640-29912677      + |      118421   LINC00161</span></span>
<span><span class="co">##      246705    chr21 29385682-29395528      + |      246705   LINC00314</span></span>
<span><span class="co">##       26046    chr21 30300466-30365277      - |       26046        LTN1</span></span>
<span><span class="co">##       29104    chr21 30244513-30257695      - |       29104      N6AMT1</span></span>
<span><span class="co">##       54088    chr21 29094698-29123552      + |       54088   LINC00113</span></span>
<span><span class="co">##        9510    chr21 28208606-28217728      - |        9510     ADAMTS1</span></span>
<span><span class="co">##                   label       col        type</span></span>
<span><span class="co">##             &lt;character&gt; &lt;integer&gt; &lt;character&gt;</span></span>
<span><span class="co">##   100616243     MIR4759         2        gene</span></span>
<span><span class="co">##       11096     ADAMTS5         5        gene</span></span>
<span><span class="co">##      118421   LINC00161         2        gene</span></span>
<span><span class="co">##      246705   LINC00314         3        gene</span></span>
<span><span class="co">##       26046        &lt;NA&gt;         6        gene</span></span>
<span><span class="co">##       29104      N6AMT1         4        gene</span></span>
<span><span class="co">##       54088   LINC00113         7        gene</span></span>
<span><span class="co">##        9510     ADAMTS1         3        gene</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 93 sequences (1 circular) from hg19 genome</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="plot-the-data-by-loopbouquetplot-function-">Plot the data by <code>loopBouquetPlot</code> function.<a class="anchor" aria-label="anchor" href="#plot-the-data-by-loopbouquetplot-function-"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">op</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">.gi</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'IMR90'</span>, <span class="st">'GM12878'</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span>  <span class="co">## plot data</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/loopBouquetPlot.html" class="external-link">loopBouquetPlot</a></span><span class="op">(</span>gi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html" class="external-link">get</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">.gi</span>, <span class="st">'_gi_sub'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  range <span class="op">=</span> <span class="va">hs_rd</span>,</span>
<span>                  feature.gr <span class="op">=</span> <span class="va">feature.gr</span>,</span>
<span>                  coor_tick_unit <span class="op">=</span> <span class="fl">5e4</span>,</span>
<span>                  coor_mark_interval <span class="op">=</span> <span class="fl">1e6</span>,</span>
<span>                  lwd.gene <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                  col.backbone <span class="op">=</span> <span class="st">'gray30'</span>,</span>
<span>                  method <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span>  <span class="co">## mark the sub-figure</span></span>
<span>  <span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">.gi</span>, x<span class="op">=</span><span class="fl">.15</span>, y<span class="op">=</span><span class="fl">.9</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span><span class="va">op</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanSample_files/figure-html/loopBouquetPlot-1.png" width="50%"><img src="HumanSample_files/figure-html/loopBouquetPlot-2.png" width="50%"></p>
<p>In the figure, the LINC00161 and LINC00314 (Long Intergenic
Non-Protein Coding RNA 161 and 314, both of them are reported in
development of bone<span class="citation"><sup>4</sup></span>) may also
be involved in the <em>ADAMTS5</em> gene regulation and expression event
for cell IMR90 but not GM12878.</p>
</div>
</div>
<div class="section level2">
<h2 id="case-2--plot-called-interactions-around-myc-gene-from-chia-pet-data">Case 2. plot called interactions around <em>MYC</em> gene from
ChIA-PET data<a class="anchor" aria-label="anchor" href="#case-2--plot-called-interactions-around-myc-gene-from-chia-pet-data"></a>
</h2>
<p>The Chromatin Interaction Analysis with Paired-End-Tag sequencing
(ChIA-PET) method encompasses chromatin immunoprecipitation (ChIP) to
enrich long-range chromatin interactions bound to a protein of interest.
Here, we visualize the chromatin interactions around the <em>MYC</em>
gene bound by RNA polymerase II (RNAPII)<span class="citation"><sup>2</sup></span> using <code>loopBouquetPlot</code>.
In this example, we highlight the effectiveness of
<code>loopBouquetPlot</code> for peak-based loops.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs_rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">'chr8:127840755-129655603'</span><span class="op">)</span></span>
<span><span class="co">## define the data file path</span></span>
<span><span class="va">extdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, </span>
<span>                       <span class="st">'GSE33664'</span>,</span>
<span>                       package<span class="op">=</span><span class="st">'loopBouquetPlotDocumentation'</span><span class="op">)</span></span>
<span><span class="va">readData</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">interaction_file</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">extdata</span>,</span>
<span>              <span class="va">interaction_file</span><span class="op">)</span>,</span>
<span>    header<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="co">## Use observed value/ expected value (donut)</span></span>
<span>  <span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">loops</span>, <span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/GInteractions-class.html" class="external-link">GInteractions</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">V1</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">V2</span><span class="op">+</span><span class="fl">1</span>, <span class="va">V3</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">V4</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">V5</span><span class="op">+</span><span class="fl">1</span>, <span class="va">V6</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     score <span class="op">=</span> <span class="va">V8</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="co">## change the chromosome names style to UCSC (with prefix 'chr')</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">loops</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'UCSC'</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">loops</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">K562_gi</span> <span class="op">&lt;-</span> <span class="fu">readData</span><span class="op">(</span><span class="st">'K562.bedpe.gz'</span><span class="op">)</span></span>
<span><span class="va">MCF7_gi</span> <span class="op">&lt;-</span> <span class="fu">readData</span><span class="op">(</span><span class="st">'MCF7.bedpe.gz'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">K562_gi_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">K562_gi</span>, <span class="va">hs_rd</span><span class="op">)</span></span>
<span><span class="va">MCF7_gi_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">MCF7_gi</span>, <span class="va">hs_rd</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## prepare the annotations</span></span>
<span><span class="va">feature.gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">hs_feature</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/interaction-accessors.html" class="external-link">regions</a></span><span class="op">(</span><span class="va">K562_gi_sub</span><span class="op">)</span>,</span>
<span>                                       <span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/interaction-accessors.html" class="external-link">regions</a></span><span class="op">(</span><span class="va">MCF7_gi_sub</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">feature.gr</span> <span class="op">&lt;-</span> <span class="va">feature.gr</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">hs_rd</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">label</span> <span class="op">&lt;-</span> <span class="va">feature.gr</span><span class="op">$</span><span class="va">symbol</span></span>
<span><span class="co">## not label the gene outside the range</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">label</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span><span class="op">&lt;</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">hs_rd</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span><span class="op">&gt;</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">hs_rd</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span><span class="co">## assign colors for the genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">7</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span>, replace<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## set the annotation type to 'gene'</span></span>
<span><span class="va">feature.gr</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="st">'gene'</span></span>
<span><span class="va">feature.gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 11 ranges and 5 metadata columns:</span></span>
<span><span class="co">##             seqnames              ranges strand |     gene_id      symbol</span></span>
<span><span class="co">##                &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">##   100302161     chr8 128972879-128972941      + |   100302161     MIR1205</span></span>
<span><span class="co">##   100302170     chr8 129021144-129021202      + |   100302170     MIR1206</span></span>
<span><span class="co">##   100302175     chr8 129061398-129061484      + |   100302175     MIR1207</span></span>
<span><span class="co">##   100302185     chr8 128808208-128808274      + |   100302185     MIR1204</span></span>
<span><span class="co">##   100302281     chr8 129162362-129162434      + |   100302281     MIR1208</span></span>
<span><span class="co">##   100750225     chr8 128025399-128033259      + |   100750225       PCAT1</span></span>
<span><span class="co">##      157638     chr8 127564683-127570711      - |      157638      LRATD2</span></span>
<span><span class="co">##        4609     chr8 128747765-128753680      + |        4609         MYC</span></span>
<span><span class="co">##        5462     chr8 128427857-128429441      + |        5462     POU5F1B</span></span>
<span><span class="co">##        5820     chr8 128806803-129113499      + |        5820        PVT1</span></span>
<span><span class="co">##      727677     chr8 128301929-128494384      - |      727677       CASC8</span></span>
<span><span class="co">##                   label       col        type</span></span>
<span><span class="co">##             &lt;character&gt; &lt;integer&gt; &lt;character&gt;</span></span>
<span><span class="co">##   100302161     MIR1205         6        gene</span></span>
<span><span class="co">##   100302170     MIR1206         7        gene</span></span>
<span><span class="co">##   100302175     MIR1207         7        gene</span></span>
<span><span class="co">##   100302185     MIR1204         2        gene</span></span>
<span><span class="co">##   100302281     MIR1208         6        gene</span></span>
<span><span class="co">##   100750225       PCAT1         2        gene</span></span>
<span><span class="co">##      157638        &lt;NA&gt;         5        gene</span></span>
<span><span class="co">##        4609         MYC         6        gene</span></span>
<span><span class="co">##        5462     POU5F1B         2        gene</span></span>
<span><span class="co">##        5820        PVT1         3        gene</span></span>
<span><span class="co">##      727677       CASC8         4        gene</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 93 sequences (1 circular) from hg19 genome</span></span></code></pre>
<div class="section level3">
<h3 id="plot-the-data-by-loopbouquetplot-function--1">Plot the data by <code>loopBouquetPlot</code> function.<a class="anchor" aria-label="anchor" href="#plot-the-data-by-loopbouquetplot-function--1"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">op</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">.gi</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'K562'</span>, <span class="st">'MCF7'</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span>  <span class="co">## plot data</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/loopBouquetPlot.html" class="external-link">loopBouquetPlot</a></span><span class="op">(</span>gi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html" class="external-link">get</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">.gi</span>, <span class="st">'_gi_sub'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  range <span class="op">=</span> <span class="va">hs_rd</span>,</span>
<span>                  feature.gr <span class="op">=</span> <span class="va">feature.gr</span>,</span>
<span>                  coor_tick_unit <span class="op">=</span> <span class="fl">5e4</span>,</span>
<span>                  coor_mark_interval <span class="op">=</span> <span class="fl">1e6</span>,</span>
<span>                  lwd.gene <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                  col.backbone <span class="op">=</span> <span class="st">'gray30'</span><span class="op">)</span></span>
<span>  <span class="co">## mark the sub-figure</span></span>
<span>  <span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">.gi</span>, x<span class="op">=</span><span class="fl">.15</span>, y<span class="op">=</span><span class="fl">.8</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span><span class="va">op</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanSample_files/figure-html/loopBouquetPlot2-1.png" width="50%"><img src="HumanSample_files/figure-html/loopBouquetPlot2-2.png" width="50%">
In the figure, the upstream genes of <em>MYC</em>, such as
<em>CASC8</em>, <em>POU5F1B</em>, and <em>PCAT1</em> etc., are released
from the complexity of RNAPII-bound-chromatin complex with
<em>CASC11</em>, <em>MYC</em>, <em>PVT1</em>, etc in cell line K562
compared to MCF7.</p>
</div>
</div>
<div class="section level2">
<h2 id="sessioninfo">sessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R Under development (unstable) (2024-03-06 r86056)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] grid      stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ChIPpeakAnno_3.37.4                    </span></span>
<span><span class="co">##  [2] rtracklayer_1.63.0                     </span></span>
<span><span class="co">##  [3] org.Hs.eg.db_3.18.0                    </span></span>
<span><span class="co">##  [4] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2</span></span>
<span><span class="co">##  [5] GenomicFeatures_1.55.3                 </span></span>
<span><span class="co">##  [6] AnnotationDbi_1.65.2                   </span></span>
<span><span class="co">##  [7] trackViewer_1.39.15                    </span></span>
<span><span class="co">##  [8] InteractionSet_1.31.0                  </span></span>
<span><span class="co">##  [9] SummarizedExperiment_1.33.3            </span></span>
<span><span class="co">## [10] Biobase_2.63.0                         </span></span>
<span><span class="co">## [11] MatrixGenerics_1.15.0                  </span></span>
<span><span class="co">## [12] matrixStats_1.2.0                      </span></span>
<span><span class="co">## [13] GenomicRanges_1.55.3                   </span></span>
<span><span class="co">## [14] GenomeInfoDb_1.39.8                    </span></span>
<span><span class="co">## [15] IRanges_2.37.1                         </span></span>
<span><span class="co">## [16] S4Vectors_0.41.4                       </span></span>
<span><span class="co">## [17] BiocGenerics_0.49.1                    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.4.0            BiocIO_1.13.0            bitops_1.0-7            </span></span>
<span><span class="co">##   [4] filelock_1.0.3           tibble_3.2.1             graph_1.81.0            </span></span>
<span><span class="co">##   [7] XML_3.99-0.16.1          rpart_4.1.23             lifecycle_1.0.4         </span></span>
<span><span class="co">##  [10] httr2_1.0.0              lattice_0.22-5           ensembldb_2.27.1        </span></span>
<span><span class="co">##  [13] MASS_7.3-60.2            backports_1.4.1          magrittr_2.0.3          </span></span>
<span><span class="co">##  [16] Hmisc_5.1-2              sass_0.4.8               rmarkdown_2.26          </span></span>
<span><span class="co">##  [19] jquerylib_0.1.4          yaml_2.3.8               plotrix_3.8-4           </span></span>
<span><span class="co">##  [22] Gviz_1.47.1              DBI_1.2.2                RColorBrewer_1.1-3      </span></span>
<span><span class="co">##  [25] abind_1.4-5              zlibbioc_1.49.0          purrr_1.0.2             </span></span>
<span><span class="co">##  [28] AnnotationFilter_1.27.0  biovizBase_1.51.0        RCurl_1.98-1.14         </span></span>
<span><span class="co">##  [31] nnet_7.3-19              VariantAnnotation_1.49.6 rappdirs_0.3.3          </span></span>
<span><span class="co">##  [34] GenomeInfoDbData_1.2.11  grImport_0.9-7           pkgdown_2.0.7           </span></span>
<span><span class="co">##  [37] codetools_0.2-19         DelayedArray_0.29.9      xml2_1.3.6              </span></span>
<span><span class="co">##  [40] tidyselect_1.2.1         futile.logger_1.4.3      universalmotif_1.21.3   </span></span>
<span><span class="co">##  [43] BiocFileCache_2.11.1     base64enc_0.1-3          GenomicAlignments_1.39.4</span></span>
<span><span class="co">##  [46] jsonlite_1.8.8           multtest_2.59.0          Formula_1.2-5           </span></span>
<span><span class="co">##  [49] survival_3.5-8           systemfonts_1.0.6        tools_4.4.0             </span></span>
<span><span class="co">##  [52] progress_1.2.3           ragg_1.2.7               strawr_0.0.91           </span></span>
<span><span class="co">##  [55] Rcpp_1.0.12              glue_1.7.0               gridExtra_2.3           </span></span>
<span><span class="co">##  [58] SparseArray_1.3.4        xfun_0.42                dplyr_1.1.4             </span></span>
<span><span class="co">##  [61] formatR_1.14             fastmap_1.1.1            latticeExtra_0.6-30     </span></span>
<span><span class="co">##  [64] rhdf5filters_1.15.2      fansi_1.0.6              digest_0.6.35           </span></span>
<span><span class="co">##  [67] R6_2.5.1                 textshaping_0.3.7        colorspace_2.1-0        </span></span>
<span><span class="co">##  [70] jpeg_0.1-10              dichromat_2.0-0.1        biomaRt_2.59.1          </span></span>
<span><span class="co">##  [73] RSQLite_2.3.5            utf8_1.2.4               tidyr_1.3.1             </span></span>
<span><span class="co">##  [76] generics_0.1.3           data.table_1.15.2        prettyunits_1.2.0       </span></span>
<span><span class="co">##  [79] httr_1.4.7               htmlwidgets_1.6.4        S4Arrays_1.3.6          </span></span>
<span><span class="co">##  [82] regioneR_1.35.0          pkgconfig_2.0.3          gtable_0.3.4            </span></span>
<span><span class="co">##  [85] blob_1.2.4               XVector_0.43.1           htmltools_0.5.7         </span></span>
<span><span class="co">##  [88] RBGL_1.79.0              ProtGenerics_1.35.3      scales_1.3.0            </span></span>
<span><span class="co">##  [91] png_0.1-8                knitr_1.45               lambda.r_1.2.4          </span></span>
<span><span class="co">##  [94] rstudioapi_0.15.0        rjson_0.2.21             checkmate_2.3.1         </span></span>
<span><span class="co">##  [97] curl_5.2.1               cachem_1.0.8             rhdf5_2.47.6            </span></span>
<span><span class="co">## [100] stringr_1.5.1            parallel_4.4.0           foreign_0.8-86          </span></span>
<span><span class="co">## [103] restfulr_0.0.15          desc_1.4.3               pillar_1.9.0            </span></span>
<span><span class="co">## [106] vctrs_0.6.5              dbplyr_2.4.0             cluster_2.1.6           </span></span>
<span><span class="co">## [109] htmlTable_2.4.2          evaluate_0.23            VennDiagram_1.7.3       </span></span>
<span><span class="co">## [112] cli_3.6.2                compiler_4.4.0           futile.options_1.0.1    </span></span>
<span><span class="co">## [115] Rsamtools_2.19.3         rlang_1.1.3              crayon_1.5.2            </span></span>
<span><span class="co">## [118] interp_1.1-6             fs_1.6.3                 stringi_1.8.3           </span></span>
<span><span class="co">## [121] deldir_2.0-4             BiocParallel_1.37.1      munsell_0.5.0           </span></span>
<span><span class="co">## [124] Biostrings_2.71.3        lazyeval_0.2.2           Matrix_1.6-5            </span></span>
<span><span class="co">## [127] BSgenome_1.71.2          hms_1.1.3                bit64_4.0.5             </span></span>
<span><span class="co">## [130] ggplot2_3.5.0            Rhdf5lib_1.25.1          KEGGREST_1.43.0         </span></span>
<span><span class="co">## [133] highr_0.10               igraph_2.0.3             memoise_2.0.1           </span></span>
<span><span class="co">## [136] bslib_0.6.1              bit_4.0.5</span></span></code></pre>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-rao20143d" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Rao,
S. S. <em>et al.</em> A 3D map of the human genome at kilobase
resolution reveals principles of chromatin looping. <em>Cell</em>
<strong>159,</strong> 1665â€“1680 (2014).</div>
</div>
<div id="ref-li2012extensive" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Li,
G. <em>et al.</em> Extensive promoter-centered chromatin interactions
provide a topological basis for transcription regulation. <em>Cell</em>
<strong>148,</strong> 84â€“98 (2012).</div>
</div>
<div id="ref-wang2016long" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Wang, Y. <em>et al.</em> Long non-coding RNA
LINC00161 sensitises osteosarcoma cells to cisplatin-induced apoptosis
by regulating the miR-645-IFIT2 axis. <em>Cancer letters</em>
<strong>382,</strong> 137â€“146 (2016).</div>
</div>
<div id="ref-shi2020long" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">Shi,
Z. <em>et al.</em> Long noncoding RNA LINC00314 facilitates osteogenic
differentiation of adipose-derived stem cells through the
hsa-miR-129-5p/GRM5 axis via the wnt signaling pathway. <em>Stem Cell
Research &amp; Therapy</em> <strong>11,</strong> 1â€“14 (2020).</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jianhong Ou.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
